src/main/java/org/jenkinsci/maven/plugins/hpi/RunMojo.java

```diff
@@ -512,17 +501,22 @@ private void generateHpl() throws MojoExecutionException {
     }
 
     @Override
-    public void configureWebApplication() throws Exception {
-        // Jetty tries to do this in WebAppContext.resolveWebApp but it failed to delete the directory.
-        File t = webApp.getTempDirectory();
-        if (t==null)    t = new File(getProject().getBuild().getDirectory(),"tmp");
-        File extractedWebAppDir= new File(t, "webapp");
+    public void configureWebApp() throws Exception {
+        if (webApp.getTempDirectory() == null) {
+            // Preëmpt AbstractWebAppMojo.configureWebApp and choose a better name
+            Path target = Paths.get(project.getBuild().getDirectory());
+            Path tmp = target.resolve("jetty");
+            if (!Files.isDirectory(tmp)) {
+                Files.createDirectories(tmp);
+            }
+            webApp.setTempDirectory(tmp.toFile());
+        }
+        File extractedWebAppDir = new File(webApp.getTempDirectory(), "webapp");
         if (isExtractedWebAppDirStale(extractedWebAppDir, webAppFile)) {
             FileUtils.deleteDirectory(extractedWebAppDir);
         }
-        
-        super.configureWebApplication();
         getWebAppConfig().setWar(webAppFile.getCanonicalPath());
+        super.configureWebApp();
```

Comment:
This has to be done after setting the WAR rather than before in order to avoid tripping an assertion.